symengine/pow.cpp

```diff
@@ -94,7 +94,12 @@ RCP<const Basic> pow(const RCP<const Basic> &a, const RCP<const Basic> &b)
         return pownum(rcp_static_cast<const Number>(b), zero);
     }
     if (eq(*b, *one)) return a;
-    if (eq(*a, *zero)) return zero;
+    if (eq(*a, *zero) && is_a_Number(*b)) {
```

Comment:
You should keep the `0**x` condition and only throw an error if `x` is a number that is negative or zero.